google_api_lib/utils.py

```diff
@@ -20,16 +26,21 @@ def __init__(self):
             settings.GOOGLE_API_AUTHN_INFO,
             scopes=settings.GOOGLE_DRIVE_PERMISSIONS_SCOPES,
         )
-        self._drive_service = googleapiclient.discovery.build(
-            "drive", "v3", credentials=self._credentials
-        )
-        self._sheets_service = googleapiclient.discovery.build(
-            "sheets", "v4", credentials=self._credentials
-        )
 
         template = (
-            self._drive_service.files()
+            self.drive_service()
+            .files()
             .get(fileId=settings.GOOGLE_SHEETS_TEMPLATE_FILE_ID, fields="owners")
             .execute()
         )
         self._sheets_owner = template["owners"][0]["emailAddress"]
+
+    def drive_service(self):
+        return googleapiclient.discovery.build(
+            "drive", "v3", credentials=self._credentials, cache_discovery=False
+        )
+
+    def sheets_service(self):
```

Comment:
I think calling sheets_service() and drive_service() will cause a new Google API client to be constructed each time the method is called. I had these initialized once in the constructor and stored as _sheets_service and _drive_service to avoid this overhead for every use. I think what we had before is preferable?

Reply:
The old code doesn't work with celery retries. I think related to some thread safety issue: https://stackoverflow.com/questions/33557857/google-datastore-ssl-errors-in-python-from-google-compute-instances

Reply:
Hmm, I see. Seems there are a lot of googleapis issues and pull requests related to similar issues. I guess this is ok for now to workaround the issue. We can revisit if we notice performance issues related to sheets operations.